### ISSUE TITLE
`Aria-label is allowed` result changes depending on the attribute order of the node

### ISSUE THREAD

**Issue by Sereza7** [validator/validator#1961]
### Summary

Hi! Thank you for improving the validator. In the recent improvements related to https://github.com/validator/validator/issues/1334 , our testing suite spotted an unexpected result:
when the role attribute is set after the `aria-label`, the parser fails. 

With the exact same DOM but the attribute order reversed, the parser doesn't fail.

AFAIK the order of attributes shouldn't impact the user experience or validity (at least in this case). For me this difference in behavior was unexpected and could be considered a bug.

### URL for document that reproduces the problem

https://www.xwikiplayground.org/xwiki/bin/view/Panels/Applications

### HTML/CSS/SVG snippet that reproduces the problem

```html
<div aria-label="Tag Cloud" class="panel
expanded PanelsTagCloud TagCloud" role="complementary"><h2 class="xwikipaneltitle">Tag Cloud</h2><div class="xwikipanelcontents"><p class="noitems">
No page has been tagged yet
</p></div></div>
```

### Checker error messages

```shell
ERROR: The “aria-label” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”. at line [1209] column [140]

Extracts of the invalid content:

  1207      <div id="xwikicontent" class="col-xs-12">
  1208                <div class="xform"><dl><dt><label>Name</label></dt><dd>Tag Cloud</dd><dt><label>Panel type</label></dt><dd>view</dd><dt><label>Category</label></dt><dd>Information</dd><dt><label>Description</label></dt><dd>
  1209  </dd><dt><label>Content</label></dt><dd><div aria-label="Tag Cloud" class="panel&#xA;expanded PanelsTagCloud TagCloud" role="complementary"><h2 class="xwikipaneltitle">Tag Cloud</h2><div class="xwikipanelcontents"><p class="noitems">
  1210  No page has been tagged yet
  1211  </p></div></div></dd><dt><label>Asynchronous rendering</label></dt><dd>
```

### Steps to reproduce

Use the `role` to make an element allowed to have an `aria-label`. Place this role after the `aria-label` in the DOM.
You can compare with the result of placing the role before the aria-label.

### Expected behavior

The DOM above does not trigger an error.

### Actual behavior

The DOM above triggers an error.

### Screenshots or screen recordings

Additional info on how we hit this: in our UI, in some places, the content goes through an additional step before rendering, that step reorders all the attributes. `aria-label` goes first and we get an unexpected report here. Everything is fine when the component is rendered on most places

**Comment by sideshowbarker** [comment]
Thanks much for catching this. I’ve fixed it now, and deployed the fix:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BPGh0bWwgbGFuZz4NCjx0aXRsZT5UZXN0PC90aXRsZT4NCjxkaXYgYXJpYS1sYWJlbD0iVGFnIENsb3VkIiBjbGFzcz0icGFuZWwNCmV4cGFuZGVkIFBhbmVsc1RhZ0Nsb3VkIFRhZ0Nsb3VkIiByb2xlPSJjb21wbGVtZW50YXJ5Ij48aDIgY2xhc3M9Inh3aWtpcGFuZWx0aXRsZSI%2BVGFnIENsb3VkPC9oMj48ZGl2IGNsYXNzPSJ4d2lraXBhbmVsY29udGVudHMiPjxwIGNsYXNzPSJub2l0ZW1zIj4NCk5vIHBhZ2UgaGFzIGJlZW4gdGFnZ2VkIHlldA0KPC9wPjwvZGl2PjwvZGl2Pg%3D%3D

**Comment by Sereza5** [comment]
Thank you very much for your quick reaction! You're amazing :D
